### PR TITLE
Support for redis 2.4

### LINKED ISSUE #185: Support redis 2.4

### PR DESCRIPTION
**Pull request by matthewloring**
Fixes #185

### CODE REVIEW

**Comment by matthewloring** [comment]
@ofrobots PTAL.

File: lib/hooks/userspace/hook-redis.js [hunk 1]
```diff
@@ -32,6 +32,23 @@ function createClientWrap(createClient) {
   };
 }
 
+// Used for redis version > 2.3
+function createStreamWrap(create_stream) {
+  return function create_stream_trace() {
+    if (!this.stream) {
+      Object.defineProperty(this, 'stream', {
```
Comment (ofrobots) [inline lib/hooks/userspace/hook-redis.js]: Why does the define property need to happen at `create_stream` invocation? Could you define the proxy by wrapping the constructor?

File: lib/hooks/userspace/hook-redis.js [hunk 2]
```diff
@@ -32,6 +32,23 @@ function createClientWrap(createClient) {
...
+        get: function () { return this._trace_stream; },
```
Comment (ofrobots) [inline lib/hooks/userspace/hook-redis.js]: The monkey-patched property needs to be a longer string, like how we do this in express, to reduce collision probability.

File: lib/hooks/userspace/hook-redis.js (hunk 1 above)
Comment (matthewloring) [inline lib/hooks/userspace/hook-redis.js]: The issue with wrapping the constructor is that before constructor invocation, the `this` object isn't accessible to my knowledge and after constructor invocation, it is already too late and create_streams has been called (https://github.com/NodeRedis/node_redis/blob/master/index.js#L109). This was the only way that I could think of to get a handle on `this` before the constructor finished executing.

File: lib/hooks/userspace/hook-redis.js (hunk 2 above)
Comment (matthewloring) [inline lib/hooks/userspace/hook-redis.js]: Done.

**Comment by matthewloring** [comment]
@ofrobots PTAL.

**Comment by ofrobots** [comment]
LGTM.

**Pull request merged**